storybookjs/storybook
# Test: Model loaders as before each and restore mocks properly

## What I did

Model loaders as before each and restore mocks properly

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

* Add console.log in preview, meta and story loaders. And check that they are run in that order, even if you gonna await something before it.
* Setup a spyOn in a play function, and see that the mock is reset when you are switching to a new story.



### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release


This pull request has been released as version [`0.0.0-pr-24948-sha-12960c76`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24948-sha-12960c76). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-24948-sha-12960c76`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24948-sha-12960c76) |
| **Triggered by** | @kasperpeulen |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`kasper/model-loaders-as-before-each`](https://github.com/storybookjs/storybook/tree/kasper/model-loaders-as-before-each) |
| **Commit** | [`12960c76`](https://github.com/storybookjs/storybook/commit/12960c76f2aaa56b01c1e9c3c13114e5c5b5542d) |
| **Datetime** | Wed Nov 22 16:24:00 UTC 2023 (`1700670240`) |
| **Workflow run** | [6960070947](https://github.com/storybookjs/storybook/actions/runs/6960070947) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=24948`_
</details>
